File: exampleProjects/PhpExample/src/de/sots/cellarsandcentaurs/domain/model/SpeedType.php
<?php

namespace de\sots\cellarsandcentaurs\domain\model;

enum SpeedType: string
{
    case WALKING = 'walking';
    case FLYING = 'flying';
    case SWIMMING = 'swimming';
    case CLIMBING = 'climbing';
    case BURROWING = 'burrowing';
}

File: exampleProjects/GoExample/src/de/sots/cellarsandcentaurs/domain/model/speed_type.go
package model

type SpeedType string

const (
	Walking   SpeedType = "WALKING"
	Flying    SpeedType = "FLYING"
	Swimming  SpeedType = "SWIMMING"
	Climbing  SpeedType = "CLIMBING"
	Burrowing SpeedType = "BURROWING"
)

func (st SpeedType) String() string {
	return string(st)
}

func (st SpeedType) IsValid() bool {
	switch st {
	case Walking, Flying, Swimming, Climbing, Burrowing:
		return true
	default:
		return false
	}
}
